R/module_teal_data.R

```diff
@@ -16,17 +16,16 @@
 #' 5. [teal_data()] object lacks any `datanames` specified in the `modules` argument.
 #'
 #' `teal` (observers in `srv_teal`) always waits to render an app until `reactive` `teal_data` is
-#' returned. If error 2-4 occurs, relevant error message is displayed to app user and after issue is
-#' resolved app will continue to run. `teal` guarantees that errors in a data don't crash an app
-#' (except error 1). This is possible thanks to `.fallback_on_failure` which returns input-data
-#' when output-data fails
-#'
+#' returned. If error 2-4 occurs, relevant error message is displayed to the app user. Once the issue is
+#' resolved, the app will continue to run. `teal` guarantees that errors in data don't crash the app
+#' (except error 1).
 #'
 #' @param id (`character(1)`) Module id
 #' @param data (`reactive teal_data`)
 #' @param data_module (`teal_data_module`)
 #' @param modules (`teal_modules` or `teal_module`) For `datanames` validation purpose
 #' @param validate_shiny_silent_error (`logical`) If `TRUE`, then `shiny.silent.error` is validated and
+#' @param is_transformer_failed (`reactiveValues`) that hosts `logical` flags of which transformers failed
 #' error message is displayed.
 #' Default is `FALSE` to handle empty reactive cycle on `init`.
```

Comment:
1. Please also include param doc for `hide_validation_error`
2. I think the param of `is_transformer_failed` was inserted in the middle of the param doc of `is_transformer_failed`.
3. doc for `is_transformer_failed` seems incomplete. We can write a little more: the index of this list of boolean is the `id` so it is easy for people to know before seeing the code.
4. The param `data_module` should be renamed to something more meaningful. For the `ui_teal_data` it takes the module UI function and for the `srv_teal_data` it takes the module server function. Perhaps it should be `data_module_ui` and `data_module_srv`